packages/perseus/src/widgets/interactive-graphs/locked-figures/locked-function.test.ts

```diff
@@ -0,0 +1,25 @@
+import {clampedDomain} from "./locked-function";
+
+describe("clampedDomain", () => {
+    test.each`
+        domain                   | graphXBounds | expected
+        ${[-Infinity, Infinity]} | ${[-10, 10]} | ${[-10, 10]}
+        ${[-20, Infinity]}       | ${[-10, 10]} | ${[-10, 10]}
+        ${[-10, Infinity]}       | ${[-10, 10]} | ${[-10, 10]}
+        ${[-5, Infinity]}        | ${[-10, 10]} | ${[-5, 10]}
+        ${[-Infinity, 20]}       | ${[-10, 10]} | ${[-10, 10]}
+        ${[-Infinity, 10]}       | ${[-10, 10]} | ${[-10, 10]}
+        ${[-Infinity, 5]}        | ${[-10, 10]} | ${[-10, 5]}
+        ${[-Infinity, -5]}       | ${[-10, 10]} | ${[-10, -5]}
+        ${[-Infinity, -10]}      | ${[-10, 10]} | ${[-10, -10]}
+        ${[10, -10]}             | ${[-10, 10]} | ${[-10, 10]}
+        ${[-10, -20]}            | ${[-10, 10]} | ${[-10, 10]}
+        ${[10, 10]}              | ${[-10, 10]} | ${[10, 10]}
+        ${[-10, 10]}             | ${[-10, 10]} | ${[-10, 10]}
```

Comment:
What happens when the `domain` doesn't overlap the `graphXBounds`? We might need to return some special value like `null` and avoid rendering the locked function at all in that case.

```suggestion
        ${[-10, 10]}             | ${[-10, 10]} | ${[-10, 10]}
        ${[-Infinity, -9]}       | ${[-10, 10]} | ${???}
```